### PR TITLE
[misc] Disable next button in push dialog if not all mandatory consents checked

### PR DESCRIPTION
**Pull request by veronikaslc**

### CODE REVIEW

**Comment by sashaandjic** [comment]
Tested: NEXT button is enabled only when all required consents are checked. 
Problem: I just noticed that push fails in the following scenario. My patient is entered before I introduced mandatory consents, and as a result I can no longer see my patient data. However, I am able to export such patient. Push option is also available, but in the very last step it fails due to the missing consents. We should disable export and push options for patients without mandatory consents. Not sure if this scenario can happen in real life...

File: components/patient-data/ui/src/main/resources/PhenoTips/PatientConsent.xml [hunk 1]
```diff
@@ -409,6 +409,14 @@
           }).get();
         },
 
+        allReqiredChecked: function() {
+          if (!this.classVariables.consentsBlock) {
+            return true;
+          }
+          var requiredUnchecked = this.classVariables.consentsBlock.find("li[class*='required'] input[type=checkbox]:not(:checked)");
```
Comment (sdumitriu) [inline components/patient-data/ui/src/main/resources/PhenoTips/PatientConsent.xml]: Why `[class*='required']`? Wouldn't `li.required` be what you actually need?

**Comment by sashaandjic** [comment]
Push is failing.

**Comment by sdumitriu** [comment]
Push works for me, but it succeeds even without mandatory consents checked.

**Comment by sashaandjic** [comment]
That's strange. I can select the fields to be pushed, but once I try to push I get this:
screen shot 2017-02-26 at 1 33 07 pm

**Comment by sashaandjic** [comment]
I retested it again with a clean install. Everything works as expected. Two things to consider fixing in the future:
- if I don't have required consent to see the patient data, I should not be able to push or export such record
- scenario: I'm trying to update patient record on a remote server by pushing it for the second time, however the patient record on the remote server no longer belong to me. The error message needs to be more explanatory. Right now it states 
image

**Comment by sashaandjic** [comment]
Not sure if we need to add "(required)" to mandatory consents. This is how it is going to look when pushing on PhenomeCentral: 
screen shot 2017-02-26 at 6 05 33 pm

**Comment by sdumitriu** [comment]
@sashaandjic Can you try pushing to the actual PhenomeCentral (or a 1.3M1 instance), perhaps that version allows pushing even without mandatory consents.

**Comment by sashaandjic** [comment]
I did. Please see the latest screenshot where I pushed the data to the preprod server. I made yet another  clean install and everything looks fine. What puzzles me is that push fails on one out 3 local instances on my machine does. Each instance comes from a different build.

File: components/patient-data/ui/src/main/resources/PhenoTips/PatientConsent.xml (hunk 1 above)
Comment (veronikaslc) [inline components/patient-data/ui/src/main/resources/PhenoTips/PatientConsent.xml]: it is

**Comment by sdumitriu** [comment]
Re-tested, "Next" is correctly disabled until all mandatory consents are checked.

@veronikaslc Can you address my codereview comments?

**Comment by veronikaslc** [comment]
on it right now

**Comment by veronikaslc** [comment]
@sdumitriu comments addressed

**Pull request merged**